File: erdjs/src/networkConfig.ts
import { IProvider } from "./interface";
import {GasPrice, GasLimit, TransactionVersion, ChainID, GasPriceModifier} from "./networkParams";

/**
 * An object holding Network configuration parameters.
 */
export class NetworkConfig {
    private static default: NetworkConfig;

    /**
     * The chain ID. E.g. "1" for the Mainnet.
     */
    public ChainID: ChainID;

    /**
     * The gas required by the Network to process a byte of the {@link TransactionPayload}.
     */
    public GasPerDataByte: number;

    /**
     *
     */
    public GasPriceModifier: GasPriceModifier;

    /**
     * The minimum gas limit required to be set when broadcasting a {@link Transaction}.
     */
    public MinGasLimit: GasLimit;

    /**
     * The minimum gas price required to be set when broadcasting a {@link Transaction}.
     */
    public MinGasPrice: GasPrice;

    /**
     * The oldest {@link TransactionVersion} accepted by the Network.
     */
    public MinTransactionVersion: TransactionVersion;

    constructor() {
        this.ChainID = new ChainID("T");
        this.GasPerDataByte = 1500;
        this.MinGasLimit = new GasLimit(50000);
        this.MinGasPrice = new GasPrice(1000000000);
        this.GasPriceModifier = new GasPriceModifier(1);
        this.MinTransactionVersion = new TransactionVersion(1);
    }

    /**
     * Gets the default configuration object (think of the Singleton pattern).
     */
    static getDefault(): NetworkConfig {
        if (!NetworkConfig.default) {
            NetworkConfig.default = new NetworkConfig();
        }

        return NetworkConfig.default;
    }

    /**
     * Synchronizes a configuration object by querying the Network, through a {@link IProvider}.
     * @param provider The provider to use
     */
    async sync(provider: IProvider): Promise<void> {
        let fresh: NetworkConfig = await provider.getNetworkConfig();
        Object.assign(this, fresh);
    }

    /**
     * Constructs a configuration object from a HTTP response (as returned by the provider).
     */
    static fromHttpResponse(payload: any): NetworkConfig {
        let networkConfig = new NetworkConfig();

        networkConfig.ChainID = new ChainID(payload["erd_chain_id"]);
        networkConfig.GasPerDataByte = Number(payload["erd_gas_per_data_byte"]);
        networkConfig.MinGasLimit = new GasLimit(payload["erd_min_gas_limit"]);
        networkConfig.MinGasPrice = new GasPrice(payload["erd_min_gas_price"]);
        networkConfig.MinTransactionVersion = new TransactionVersion(payload["erd_min_transaction_version"]);
        networkConfig.GasPriceModifier = new GasPriceModifier(payload["erd_gas_price_modifier"]);

        return networkConfig;
    }
}

File: erdjava/src/main/java/elrond/NetworkConfig.java
package elrond;

import java.io.IOException;

import elrond.Exceptions.ErrProxyRequest;
import elrond.ProxyProvider.PayloadOfGetNetworkConfig;

public class NetworkConfig {
    private static final NetworkConfig defaultInstance = new NetworkConfig();

    private String chainID;
    private int gasPerDataByte;
    private long minGasLimit;
    private long minGasPrice;
    private int minTransactionVersion;

    public NetworkConfig() {
        this.chainID = "T";
        this.gasPerDataByte = 1500;
        this.minGasLimit = 50000;
        this.minGasPrice = 1000000000;
        this.minTransactionVersion = 1;
    }

    public static NetworkConfig getDefault() {
        return defaultInstance;
    }

    /**
     * Synchronizes the object with the current configuration from a provider
     * 
     * @param provider
     * @throws IOException
     * @throws ErrProxyRequest
     */
    public void sync(IProvider provider) throws IOException, ErrProxyRequest {
        NetworkConfig fresh = provider.getNetworkConfig();
        this.assign(fresh);
    }

    private void assign(NetworkConfig other) {
        this.chainID = other.chainID;
        this.gasPerDataByte = other.gasPerDataByte;
        this.minGasLimit = other.minGasLimit;
        this.minGasPrice = other.minGasPrice;
        this.minTransactionVersion = other.minTransactionVersion;
    }

    public static NetworkConfig fromProviderPayload(PayloadOfGetNetworkConfig response) {
        NetworkConfig config = new NetworkConfig();

        config.chainID = response.chainID;
        config.gasPerDataByte = response.gasPerDataByte;
        config.minGasLimit = response.minGasLimit;
        config.minGasPrice = response.minGasPrice;
        config.minTransactionVersion = response.minTransactionVersion;

        return config;
    }

    public String getChainID() {
        return chainID;
    }

    public void setChainID(String chainID) {
        this.chainID = chainID;
    }

    public int getGasPerDataByte() {
        return gasPerDataByte;
    }

    public void setGasPerDataByte(int gasPerDataByte) {
        this.gasPerDataByte = gasPerDataByte;
    }

    public long getMinGasLimit() {
        return minGasLimit;
    }

    public void setMinGasLimit(long minGasLimit) {
        this.minGasLimit = minGasLimit;
    }

    public long getMinGasPrice() {
        return minGasPrice;
    }

    public void setMinGasPrice(long minGasPrice) {
        this.minGasPrice = minGasPrice;
    }

    public int getMinTransactionVersion() {
        return minTransactionVersion;
    }

    public void setMinTransactionVersion(int minTransactionVersion) {
        this.minTransactionVersion = minTransactionVersion;
    }
}